### ISSUE TITLE
[v851s] Linux kernel patch

### ISSUE THREAD

**Issue by kamejoko80** [szemzoa/awboot#23]
Hi @szemzoa 

Thanks for distributing a great bootloader for Allwinner SoCs, I'd like to try a new Linux kernel with v851s (Yuzukilizard), would you please share the patch files?

**Comment by szemzoa** [comment]
Yes, just give me a few days, please

**Comment by szemzoa** [comment]
I uploaded it. Please note that I'm sure there are a lot of things that not correct in these patches...

**Comment by kamejoko80** [comment]
@szemzoa 

Thanks for your quick support, I will try and I hope I can fix the problem if any.

**Comment by kamejoko80** [comment]
Hi @szemzoa,

When applying your patch file I saw error below:

```
v851s-lizard-linux.patch:322: trailing whitespace.
                        interrupts = <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>, /* GPIOA */
v851s-lizard-linux.patch:323: trailing whitespace.
                                     <GIC_SPI 71 IRQ_TYPE_LEVEL_HIGH>, /* GPIOC */
v851s-lizard-linux.patch:324: trailing whitespace.
                                     <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>, /* GPIOD */
v851s-lizard-linux.patch:325: trailing whitespace.
                                     <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>, /* GPIOE */
v851s-lizard-linux.patch:326: trailing whitespace.
                                     <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>, /* GPIOF */
error: affected file 'scripts/dtc/include-prefixes/arm/allwinner/Makefile' is beyond a symbolic link
error: affected file 'scripts/dtc/include-prefixes/arm/allwinner/sun8i-v851s-lizard.dts' is beyond a symbolic link
error: affected file 'scripts/dtc/include-prefixes/arm/allwinner/sun8i-v851s.dtsi' is beyond a symbolic link
error: affected file 'scripts/dtc/include-prefixes/dt-bindings/clock/sun8i-v85x-ccu.h' is beyond a symbolic link
error: affected file 'scripts/dtc/include-prefixes/dt-bindings/reset/sun8i-v85x-ccu.h' is beyond a symbolic link
```

Then I fixed the patch file (see attachment) and added Foresee SPI NAND flash to boot with UBIFS.
Finally, my board can boot properly.

Thank you very much.

[v851s-lizard-linux-v6.6-rc1.patch](https://github.com/szemzoa/awboot/files/12643442/v851s-lizard-linux-v6.6-rc1.patch)
[v851s-lizard-linux-v6.6-rc1-bootlog.txt](https://github.com/szemzoa/awboot/files/12643443/v851s-lizard-linux-v6.6-rc1-bootlog.txt)

**Comment by szemzoa** [comment]
Thanks, I uploaded your updated patch, and also I have added a compatible what I forgot to make USB working.

**Comment by kamejoko80** [comment]
Hi @szemzoa,

I've applied your new patch, however the usb otg still doesn't work 

```
[   11.371427] platform 4100000.usb: deferred probe pending
[   11.376876] platform 4101000.usb: deferred probe pending
[   11.382196] platform 4101400.usb: deferred probe pending
```

Is there any miss-configuration?

**Comment by kamejoko80** [comment]
Hi @szemzoa,

I've tried to bring up the mipi-dsi display, just added some missing lock domains but I don't have any clue, would you mind sharing some hints? 
([v851s-lizard-linux-v6.6-rc1-mipi-dsi.patch](https://github.com/szemzoa/awboot/files/12721789/v851s-lizard-linux-v6.6-rc1-mipi-dsi.patch))

**Comment by szemzoa** [comment]
I don't think this is the best place for kernel related questions. Why don't you try sunxi IRC channel? There are a lot of people with much better knowledge about mainline linux: https://linux-sunxi.org/IRC

**Comment by kamejoko80** [comment]
Hi @szemzoa,

Thanks for your quick response, I'll reach out to them.

**Comment by szemzoa** [comment]
The dsi clocks should be:                         
clocks = <&ccu CLK_BUS_MIPI_DSI>,
               <&ccu CLK_TCON_TOP_DSI>;

and try something like in the board .dts:

&de {
        status = "okay";
};
&dphy {
        status = "okay";
};
&tcon_lcd0 {
        status = "okay";
};
&dsi {
        status = "okay";

        pinctrl-0 = <&dsi_2lane_pins>;
        pinctrl-names = "default";

        ports {
            #address-cells = <1>;
            #size-cells = <0>;

            dsi_out: port@0 {
                reg = <0>;

                dsi_out_panel: endpoint {
                    remote-endpoint = <&panel_out_dsi>;
                };
            };
        };

        panel@0 {
            compatible = "lg,lh500wx1-sd03";
            reg = <0>;
/*
            backlight = <&backlight>;
*/
            port {
                panel_out_dsi: endpoint {
                    remote-endpoint = <&dsi_out_panel>;
                };
            };
        };
};

I don't have MIPI panel, but I see the whole pipeline is registered(also, cmd line drm.debug=0x3 can be usefull):

[    1.244268] sun4i-drm display-engine: bound 5100000.mixer (ops 0x40858378)
[    1.251651] sun4i-drm display-engine: bound 5460000.tcon-top (ops 0x4085c858)
[    1.259521] sun4i-drm display-engine: No panel or bridge found... RGB output disabled
[    1.267377] sun4i-drm display-engine: bound 5461000.lcd-controller (ops 0x40855708)
[    1.275194] sun4i-drm display-engine: bound 5450000.dsi (ops 0x40857ee0)
[    1.282246] [drm:drm_minor_register] 
[    1.282288] [drm:drm_minor_register] 
[    1.282797] [drm:drm_minor_register] new minor registered 0
[    1.282839] [drm:drm_minor_register] 
[    1.282913] [drm:drm_sysfs_connector_add] adding "DSI-1" to sysfs
[    1.283158] [drm] Initialized sun4i-drm 1.0.0 20150629 for display-engine on minor 0
[    1.291052] [drm:drm_mode_object_get] OBJ ID: 43 (2)
[    1.291147] [drm:drm_sysfs_hotplug_event] generating hotplug event
[    1.291203] [drm:drm_mode_object_put] OBJ ID: 43 (2)
[    1.291231] sun4i-drm display-engine: [drm] Cannot find any crtc or sizes
[    1.298182] [drm:drm_mode_object_get] OBJ ID: 43 (2)
[    1.298230] [drm:drm_mode_object_put] OBJ ID: 43 (2)
[    1.298251] sun4i-drm display-engine: [drm] Cannot find any crtc or sizes
